File: ios/LndMobile/LndMobileScheduledSync.m
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(LndMobileScheduledSync, NSObject)

RCT_EXTERN_METHOD(
  setupScheduledSyncWork: (RCTPromiseResolveBlock)resolve
  rejecter: (RCTPromiseRejectBlock)reject
)

RCT_EXTERN_METHOD(
  removeScheduledSyncWork: (RCTPromiseResolveBlock)resolve
  rejecter: (RCTPromiseRejectBlock)reject
)

RCT_EXTERN_METHOD(
  checkScheduledSyncWorkStatus: (RCTPromiseResolveBlock)resolve
  rejecter: (RCTPromiseRejectBlock)reject
)

@end

File: ios/LndMobile/LndMobileScheduledSync.swift
import Foundation

@objc(LndMobileScheduledSync)
class LndMobileScheduledSync: NSObject, RCTBridgeModule {
  @objc
  static func moduleName() -> String! {
    "LndMobileScheduledSync"
  }

  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }

  @objc(setupScheduledSyncWork:rejecter:)
  func setupScheduledSyncWork(resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock ) -> Void {
    resolve(true)
  }

  @objc(removeScheduledSyncWork:rejecter:)
  func removeScheduledSyncWork(resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) -> Void {
    resolve(true)
  }

  @objc(checkScheduledSyncWorkStatus:rejecter:)
  func checkScheduledSyncWorkStatus(resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) -> Void {
  }
}
